data/locus_coeruleus/locus_coeruleus.R

```diff
@@ -0,0 +1,144 @@
+#!/usr/bin/env Rscript
+
+# Author_and_contribution: Niklas Mueller-Boetticher; created template
+# Author_and_contribution: Florian Heyl (@heylf); created code
+
+# H_E.json and H_E.tiff not public. Request for access is still unanswered.
+
+suppressPackageStartupMessages(library(optparse))
+suppressPackageStartupMessages(library(SpatialExperiment))
+suppressPackageStartupMessages(library(SingleCellExperiment))
+suppressPackageStartupMessages(library(WeberDivechaLCdata))
+suppressPackageStartupMessages(library(Matrix))
+
+option_list <- list(
+  make_option(
+    c("-o", "--out_dir"),
+    type = "character", default = NULL,
+    help = "Output directory to write files to."
+  )
+)
+
+description <- "Load data (Visium) for Locus_coeruleus from Lukas M. Weber at al. (2022); 
+The gene expression landscape of the human locus coeruleus revealed 
+by single-nucleus and spatially-resolved transcriptomics"
+
+opt_parser <- OptionParser(
+  usage = description,
+  option_list = option_list
+)
+opt <- parse_args(opt_parser)
+
+out_dir <- opt$out_dir
+
+args = commandArgs(trailingOnly=TRUE)
+
+# Load data
+spe <- WeberDivechaLCdata_Visium()
+
+coords <- as.data.frame(spe@int_colData$spatialCoords)
+colnames(coords) <- c('x', 'y')
+
+counts <- spe@assays@data$counts
+
+counts_func <- function(x){
+  fields <- unlist(strsplit(x, "_"))
+  if ( length(fields) == 4 ){
+    return(paste(fields[1:3], collapse = "_"))
+  } else {
+    return(paste(fields[1:4], collapse = "_"))
+  }
+}
+counts_lc <- unlist(lapply(colnames(counts), counts_func))
+
+obs_func <- function(x){
+  fields <- unlist(strsplit(x, "_"))
+  if ( length(fields) == 4 ){
+    return(fields[4])
+  } else{
+    return(fields[5])
+  }
+}
+observations_lc <- unlist(lapply(colnames(counts), obs_func))
+
+LC_samples <- unique(spe@colData$sample_id)
+
+out_dir <- 'out_dir'
+for ( dir in LC_samples ){
+    dir <- paste0(out_dir, "/", dir)
+    if ( dir.exists(dir) == FALSE ){
+        dir.create(dir, recursive=TRUE)
+    }
+}
+
+print("Create output ...")
+
+patient_list <- c()
+sample_list <- c()
+directory_list <- c()
+
+# Write coordinates.tsv, observations.tsv, features.tsv, counts.mtx and labels.tsv
+for (lc in LC_samples){
+    
+    print(lc)
+    dir <- paste0(out_dir, "/", lc)
+        # TODO Please check this again beause I had to flip the coordinates to get the pictures as seen in:
```

Comment:
Couldn't open the link. Why do we want to flip the coordinate since H&E images are not available so far?